bench/Variants.hs

```diff
@@ -0,0 +1,170 @@
+module Variants where
+
+import Base
+import Development.Shake hiding ((<//>))
+import Juvix.Prelude
+import Juvix.Prelude.Env
+
+defaultVariants :: [Variant]
+defaultVariants =
+  [ ocamlExe,
+    ocamlByteExe,
+    juvixExe,
+    -- juvixWasm,
+    clangExe,
+    -- clangWasm,
+    runtimeExe,
+    -- runtimeWasm,
+    haskellExe,
+    haskellStrictExe
+  ]
+
+defaultExt :: [String]
+defaultExt = [".exe"]
+
+outOptions :: BuildArgs -> [String] -> [String]
+outOptions args ext = ["-o", toFilePath (binFile args ext)]
+
+commonOptions :: BuildArgs -> [String] -> [String]
+commonOptions args ext = toFilePath (args ^. buildSrc) : outOptions args ext
+
+ocamlExe :: Variant
+ocamlExe =
+  Variant
+    { _variantTitle = Nothing,
+      _variantLanguage = Ocaml,
+      _variantExtensions = ext,
+      _variantBuild = \args ->
+        command_ [] "ocamlc" (commonOptions args ext)
+    }
+  where
+    ext :: [String]
+    ext = defaultExt
+
+ocamlByteExe :: Variant
+ocamlByteExe =
+  Variant
+    { _variantTitle = Just "byte",
+      _variantLanguage = Ocaml,
+      _variantExtensions = ext,
+      _variantBuild = \args ->
+        command_ [] "ocamlopt" ("-O2" : commonOptions args ext)
```

Comment:
`ocamlopt` creates a native executable not bytecode

Reply:
swapped